lib/commands/favorite.go

```diff
@@ -43,12 +82,29 @@ func (c *Cmd) ListFavorites(cCtx *cli.Context) error {
 // argument it is used, otherwise the user is walked through a wizard to make a
 // selection.
 func (c *Cmd) Favorites(cCtx *cli.Context) error {
-	// map our favorites for ease of use
-	fNames, fMap := helper.MapFavs(c.config.Favorites)
+
+	// get the combined list of favorites from the CLI config and the Kion API (if compatible)
+	useApi := cCtx.App.Metadata["useFavoritesAPI"].(bool)
+	var apiFavorites []structs.Favorite
+	var err error
+	if useApi {
+		apiFavorites, _, err = kion.GetAPIFavorites(c.config.Kion.Url, c.config.Kion.ApiKey)
+		if err != nil {
+			fmt.Printf("Error retrieving favorites from API: %v\n", err)
+			return nil
+		}
+	}
+	result, err := helper.CombineFavorites(c.config.Favorites, apiFavorites)
+	if err != nil {
+		fmt.Printf("Error combining favorites: %v\n", err)
+		return nil
```

Comment:
This should return the error instead of nil. The command will bubble it all the way up to exit.